### PR TITLE
SUS-2038 | AllUsers API returning incorrect results for user groups

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/SUS-2038

In the docs, it allows multiple 'augroup's to be listed and separated by a pipe character. However, it only reads the first group

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues
* ![MAJOR][MAJOR] 16 major
* ![MINOR][MINOR] 1 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L104](https://github.com/Wikia/app/blob/56373ece083e3ad217501b91c0f42f5127b547bb/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L104): Remove this unused "$wgMemc" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L170](https://github.com/Wikia/app/blob/56373ece083e3ad217501b91c0f42f5127b547bb/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L170): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L212](https://github.com/Wikia/app/blob/56373ece083e3ad217501b91c0f42f5127b547bb/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L212): Remove this unused "$data" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L272](https://github.com/Wikia/app/blob/56373ece083e3ad217501b91c0f42f5127b547bb/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L272): The Cyclomatic Complexity of this function "local_users" is 24 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L294](https://github.com/Wikia/app/blob/56373ece083e3ad217501b91c0f42f5127b547bb/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L294): Remove this unused "$fld_blockinfo" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L295](https://github.com/Wikia/app/blob/56373ece083e3ad217501b91c0f42f5127b547bb/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L295): Remove this unused "$fld_editcount" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L296](https://github.com/Wikia/app/blob/56373ece083e3ad217501b91c0f42f5127b547bb/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L296): Remove this unused "$fld_groups" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L297](https://github.com/Wikia/app/blob/56373ece083e3ad217501b91c0f42f5127b547bb/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L297): Remove this unused "$fld_registration" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L299](https://github.com/Wikia/app/blob/56373ece083e3ad217501b91c0f42f5127b547bb/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L299): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L302](https://github.com/Wikia/app/blob/56373ece083e3ad217501b91c0f42f5127b547bb/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L302): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues
* ![MAJOR][MAJOR] 16 major
* ![MINOR][MINOR] 1 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L109](https://github.com/Wikia/app/blob/560f86c5482e9c1fd9bcc5313865f2828732c7b5/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L109): Remove this unused "$wgMemc" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L175](https://github.com/Wikia/app/blob/560f86c5482e9c1fd9bcc5313865f2828732c7b5/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L175): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L217](https://github.com/Wikia/app/blob/560f86c5482e9c1fd9bcc5313865f2828732c7b5/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L217): Remove this unused "$data" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L277](https://github.com/Wikia/app/blob/560f86c5482e9c1fd9bcc5313865f2828732c7b5/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L277): The Cyclomatic Complexity of this function "local_users" is 24 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L299](https://github.com/Wikia/app/blob/560f86c5482e9c1fd9bcc5313865f2828732c7b5/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L299): Remove this unused "$fld_blockinfo" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L300](https://github.com/Wikia/app/blob/560f86c5482e9c1fd9bcc5313865f2828732c7b5/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L300): Remove this unused "$fld_editcount" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L301](https://github.com/Wikia/app/blob/560f86c5482e9c1fd9bcc5313865f2828732c7b5/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L301): Remove this unused "$fld_groups" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L302](https://github.com/Wikia/app/blob/560f86c5482e9c1fd9bcc5313865f2828732c7b5/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L302): Remove this unused "$fld_registration" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L304](https://github.com/Wikia/app/blob/560f86c5482e9c1fd9bcc5313865f2828732c7b5/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L304): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L307](https://github.com/Wikia/app/blob/560f86c5482e9c1fd9bcc5313865f2828732c7b5/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L307): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by mszabo-wikia — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues
* ![MAJOR][MAJOR] 16 major
* ![MINOR][MINOR] 1 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L78](https://github.com/Wikia/app/blob/a72378859e1aaa6f61141ef94c591a31f4acdc39/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L78): Remove this unused "$wgMemc" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L144](https://github.com/Wikia/app/blob/a72378859e1aaa6f61141ef94c591a31f4acdc39/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L144): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L184](https://github.com/Wikia/app/blob/a72378859e1aaa6f61141ef94c591a31f4acdc39/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L184): Remove this unused "$data" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L246](https://github.com/Wikia/app/blob/a72378859e1aaa6f61141ef94c591a31f4acdc39/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L246): The Cyclomatic Complexity of this function "local_users" is 24 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L268](https://github.com/Wikia/app/blob/a72378859e1aaa6f61141ef94c591a31f4acdc39/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L268): Remove this unused "$fld_blockinfo" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L269](https://github.com/Wikia/app/blob/a72378859e1aaa6f61141ef94c591a31f4acdc39/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L269): Remove this unused "$fld_editcount" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L270](https://github.com/Wikia/app/blob/a72378859e1aaa6f61141ef94c591a31f4acdc39/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L270): Remove this unused "$fld_groups" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L271](https://github.com/Wikia/app/blob/a72378859e1aaa6f61141ef94c591a31f4acdc39/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L271): Remove this unused "$fld_registration" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L273](https://github.com/Wikia/app/blob/a72378859e1aaa6f61141ef94c591a31f4acdc39/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L273): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [WikiaApiQueryAllUsers.php#L276](https://github.com/Wikia/app/blob/a72378859e1aaa6f61141ef94c591a31f4acdc39/extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php#L276): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 1]
```diff
@@ -260,9 +227,11 @@ private function global_users() {
 					$lastUserData['registration'] = wfTimestamp(TS_ISO_8601, $row->user_registration);
 				}
 
-			// Add user's group info
+				// Add user's group info
 				if ( $this->fld_groups ) {
-					$lastUserData['groups'] = array_key_exists($row->user_id, $user_groups) ? $user_groups[$row->user_id] : array();
+					$lastUserData['groups'] = ( !empty( $users ) && array_key_exists($row->user_id, $users) )
```
Comment (mszabo-wikia) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php]: Let's be spacey here

**Review by mszabo-wikia — APPROVED** [review]

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 2]
```diff
@@ -263,55 +251,55 @@ private function local_users() {
...
 
-		$this->addFields('user_id, user_name, cnt_groups, edits as user_editcount, all_groups, user_is_blocked');
-		$this->addOption('ORDER BY', 'user_name');
+		$this->addFields( 'user_id, user_name, cnt_groups, edits as user_editcount, all_groups, user_is_blocked' );
+		$this->addOption( 'ORDER BY', 'user_name' );
 
-		$res = $this->select(__METHOD__);
+		$res = $this->select( __METHOD__ );
 
 		$data = array ();
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:291]: MAJOR Remove this unused "$data" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 3]
```diff
@@ -345,36 +333,36 @@ private function local_users() {
...
 							$lastUserData['groups'][] = $group;
 						}
 						$use_group[] = $group;
 					}
 				}
-						
-				if ( isset($lastUserData['groups']) ) 
-					$result->setIndexedTagName($lastUserData['groups'], 'g');
+
+				if ( isset( $lastUserData['groups'] ) )
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:358]: MAJOR Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 4]
```diff
@@ -320,23 +308,23 @@ private function local_users() {
...
+
 				# is blocked
-				if ($this->fld_blockinfo) {
+				if ( $this->fld_blockinfo ) {
 					$blocker_name = $block_reason = '';
 					if ( $row->user_is_blocked ) {
-						$oBlock = Block::newFromTarget( User::whoIs($row->user_id), 0 ); 
-						if ( is_object($oBlock) ) {
+						$oBlock = Block::newFromTarget( User::whoIs( $row->user_id ), 0 );
+						if ( is_object( $oBlock ) ) {
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:327]: MAJOR Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 5]
```diff
@@ -263,55 +251,55 @@ private function local_users() {
...
-		$fld_registration 	= isset($prop['registration']);
-
-		if ( !is_null($params['from']) )
-			$this->addWhere( 'user_name >= ' . $db->addQuotes($this->keyToTitle($params['from'])) );
-
-		if ( !is_null($params['prefix']) )
+		$this->addOption( 'LIMIT', $limit );
+
+		# params
+		$fld_blockinfo 		= isset( $prop['blockinfo'] );
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:268]: MAJOR Remove this unused "$fld_blockinfo" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 6]
```diff
@@ -263,55 +251,55 @@ private function local_users() {
...
+		$fld_editcount 		= isset( $prop['editcount'] );
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:269]: MAJOR Remove this unused "$fld_editcount" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 7]
```diff
@@ -13,130 +13,118 @@
...
 			$this->profileDBOut();
 		}
 
 		wfProfileOut( __METHOD__ );
 		return $data;
 	}
 
-	private function getUserRegistration($user_id) {
+	private function getUserRegistration( $user_id ) {
 		global $wgMemc, $wgExternalSharedDB;
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:78]: MAJOR Remove this unused "$wgMemc" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 8]
```diff
@@ -263,55 +251,55 @@ private function local_users() {
...
-		if ( !is_null($params['from']) )
-			$this->addWhere( 'user_name >= ' . $db->addQuotes($this->keyToTitle($params['from'])) );
-
-		if ( !is_null($params['prefix']) )
+		$this->addOption( 'LIMIT', $limit );
+
+		# params
+		$fld_blockinfo 		= isset( $prop['blockinfo'] );
+		$fld_editcount 		= isset( $prop['editcount'] );
+		$fld_groups 		= isset( $prop['groups'] );
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:270]: MAJOR Remove this unused "$fld_groups" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 9]
```diff
@@ -263,55 +251,55 @@ private function local_users() {
...
+		$fld_registration 	= isset( $prop['registration'] );
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:271]: MAJOR Remove this unused "$fld_registration" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 10]
```diff
@@ -149,15 +137,15 @@ private function global_users() {
...
 
 		$limit = $this->params['limit'];
-		$this->addTables('`user`', 'u1');
+		$this->addTables( '`user`', 'u1' );
 
 		if ( !is_null( $this->params['group'] ) ) {
 			$users = $this->getUsersForGroup();
-			if ( empty($users) ) 
-				$users = array(0);
+			if ( empty( $users ) )
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:144]: MAJOR Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 11]
```diff
@@ -263,55 +251,55 @@ private function local_users() {
...
-		if ( !is_null($params['prefix']) )
+		$this->addOption( 'LIMIT', $limit );
+
+		# params
+		$fld_blockinfo 		= isset( $prop['blockinfo'] );
+		$fld_editcount 		= isset( $prop['editcount'] );
+		$fld_groups 		= isset( $prop['groups'] );
+		$fld_registration 	= isset( $prop['registration'] );
+
+		if ( !is_null( $params['from'] ) )
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:273]: MAJOR Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 12]
```diff
@@ -263,55 +251,55 @@ private function local_users() {
...
+			$this->addWhere( 'user_name >= ' . $db->addQuotes( $this->keyToTitle( $params['from'] ) ) );
+
+		if ( !is_null( $params['prefix'] ) )
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:276]: MAJOR Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 13]
```diff
@@ -170,89 +158,89 @@ private function global_users() {
...
 		}
 
-		$db->freeResult($res);
+		$db->freeResult( $res );
 
-		$result->setIndexedTagName_internal(array('query', $this->getModuleName()), 'u');
+		$result->setIndexedTagName_internal( array( 'query', $this->getModuleName() ), 'u' );
 	}
 
 	private function local_users() {
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:246]: MAJOR The Cyclomatic Complexity of this function "local_users" is 24 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 14]
```diff
@@ -320,23 +308,23 @@ private function local_users() {
 				}
 
 				// No more rows left
-				if (!$row) break;
+				if ( !$row ) break;
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:311]: MAJOR Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
MINOR 2 statements were found on this line. Reformat the code to have only one statement per line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS122)

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 15]
```diff
@@ -170,89 +158,89 @@ private function global_users() {
...
 		$this->addFields  ( 'u1.user_name, u1.user_id' );
 		$this->addFieldsIf( 'u1.user_editcount', $this->fld_editcount );
 		$this->addFieldsIf( 'u1.user_registration', $this->fld_registration );
 
 		$this->addOption  ( 'ORDER BY', 'u1.user_name' );
 
-		$res = $this->select(__METHOD__);
+		$res = $this->select( __METHOD__ );
 
 		$data = array ();
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:184]: MAJOR Remove this unused "$data" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 16]
```diff
@@ -263,55 +251,55 @@ private function local_users() {
...
+
+		if ( !is_null( $params['prefix'] ) )
 			$this->addWhere( 'user_name' . $db->buildLike( $this->keyToTitle( $params['prefix'] ), $db->anyString() ) );
 
-		if ( !is_null($params['group']) ) {
+		if ( !is_null( $params['group'] ) ) {
 			$this->addWhere( 'all_groups' . $db->buildLike( $db->anyString(), $params['group'], $db->anyString() ) );
 		}
 
 		if ( $params['witheditsonly'] )
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:283]: MAJOR Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php [hunk 17]
```diff
@@ -345,36 +333,36 @@ private function local_users() {
...
 
 			// Add user's group info
 			if ( $this->fld_groups && $row->cnt_groups > 0 ) {
 				$use_group = array();
-				$groups = explode(";", $row->all_groups);
-				if ( !empty($groups) ) {
+				$groups = explode( ";", $row->all_groups );
+				if ( !empty( $groups ) ) {
 					foreach ( $groups as $group ) {
 						if ( !in_array( $group, $use_group ) ) {
```
Comment (WikiaTech) [inline extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php:351]: MAJOR Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues
* ![MAJOR][MAJOR] 16 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php (hunk 1 above)
Comment (macbre) [reply extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php]: Run `stylize.php`

**Pull request merged**